libs/domains/variables/feature/src/lib/output-variables/output-variables.tsx

```diff
@@ -0,0 +1,192 @@
+        <button type="button" className="w-4 text-brand-500" onClick={() => setVisible((visible) => !visible)}>
+          {visible ? <Icon name={IconAwesomeEnum.EYE_SLASH} /> : <Icon name={IconAwesomeEnum.EYE} />}
+        </button>
+      </Tooltip>
+      {visible ? (
+        <>
+          <span className="text-brand-500">{variable.value}</span>
+          {Boolean(variable.value) && (
+            <Tooltip content="Copy">
+              <button
+                className="text-brand-500"
+                onClick={() => variable.value !== null && onClickCopyToClipboard(variable.value)}
+              >
+                <Icon name={icon} />
+              </button>
+            </Tooltip>
+          )}
+        </>
+      ) : (
+        <span className="text-neutral-350 text-2xl font-medium pt-1.5">*************</span>
+      )}
+    </span>
+  )
+}
+
+interface JobOutputVariablesProps extends ComponentProps<typeof Table.Root> {
+  serviceId: string
+}
+
+export function OutputVariables({ serviceId, className, ...props }: JobOutputVariablesProps) {
+  const { data = [] } = useVariables({ parentId: serviceId, scope: 'JOB' })
+  const [sorting, setSorting] = useState<SortingState>([])
+  const keyPrefix = `QOVERY_OUTPUT_JOB_Z${serviceId.split('-')[0].toUpperCase()}_`
+
+  const variables: Variable[] = useMemo(() => {
+    return data
+      .filter(({ key }) => key.startsWith(keyPrefix))
+      .map((v) => ({ ...v, key: v.key.slice(keyPrefix.length) }))
+  }, [data])
+
+  const columnHelper = createColumnHelper<Variable>()
+  const columns = useMemo(
+    () => [
+      columnHelper.accessor('key', {
+        header: () => (
+          <>
+            Job output variables
+            <Tooltip
+              content={
+                <>
+                  The Job output is injected as environment variables <br />
+                  to any service within this environment
+                </>
+              }
+            >
+              <span>
+                <Icon name={IconAwesomeEnum.CIRCLE_INFO} className="text-neutral-350" />
+              </span>
+            </Tooltip>
+          </>
+        ),
+        enableSorting: true,
+        size: 50,
+        cell: (info) => {
+          const key = info.getValue()
+          return <span className="text-sm text-neutral-350">{key}</span>
+        },
+      }),
+      columnHelper.accessor('value', {
+        header: 'Value',
+        enableSorting: false,
+        size: 50,
+        cell: (info) => {
+          return <VariableValue variable={info.row.original} />
+        },
+      }),
+    ],
+    []
+  )
+
+  const table = useReactTable({
+    data: variables,
+    columns,
+    state: {
+      sorting,
+    },
+    onSortingChange: setSorting,
+    getCoreRowModel: getCoreRowModel(),
+    getRowCanExpand: () => true,
+    getExpandedRowModel: getExpandedRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+  })
+
+  return (
```

Comment:
When it's loaded and the data is empty could you return null ? it's to avoid this case 
<img width="1236" alt="Capture d’écran 2024-02-15 à 11 29 35" src="https://github.com/Qovery/console/assets/533928/22914db7-84a3-45a8-adcf-4a23f79fb687">

I have this case with cronjob

Reply:
Fixed https://github.com/Qovery/console/pull/1177/commits/2cd646cff22337ba2e91411a08b943e5aece4feb